test/codegen-generated-classes-test/src/test/java/software/amazon/awssdk/services/CredentialScopeTest.java

```diff
@@ -0,0 +1,163 @@
+            System.setProperty("aws.secretAccessKey", "skid");
+            System.setProperty("aws.credentialScope", "us-west-1");
+        }
+        if (!clientLevel && profileLevel) {
+            clientBuilder.credentialsProvider(DefaultCredentialsProvider
+                                                  .builder()
+                                                  .profileFile(credentialFile("test", "access", "secret","us-west-2"))
+                                                  .profileName("test")
+                                                  .build());
+        }
+
+        ProtocolQueryClient client = clientBuilder.build();
+        OperationWithNoInputOrOutputRequest.Builder requestBuilder = OperationWithNoInputOrOutputRequest.builder();
+        if (requestLevel) {
+            requestBuilder.overrideConfiguration(o -> o.credentialsProvider(StaticCredentialsProvider.create(AwsBasicCredentials.builder()
+                                                                                                                                .accessKeyId("akid")
+                                                                                                                                .secretAccessKey("skid")
+                                                                                                                                .credentialScope("us-east-1")
+                                                                                                                                .build())));
+        }
+
+        OperationWithNoInputOrOutputRequest request = requestBuilder.build();
+        stubFor(post(anyUrl()).willReturn(aResponse().withStatus(200)));
+        client.operationWithNoInputOrOutput(request);
+
+        Region credentialScope = capturingInterceptor.credentialScope;
+        if (requestLevel) {
+            assertThat(credentialScope).isEqualTo(Region.US_EAST_1);
+        } else if (clientLevel) {
+            assertThat(credentialScope).isEqualTo(Region.US_EAST_2);
+        } else if (envVarLevel) {
+            assertThat(credentialScope).isEqualTo(Region.US_WEST_1);
+        } else if (profileLevel) {
+            assertThat(credentialScope).isEqualTo(Region.US_WEST_2);
+        } else {
+            assertThat(credentialScope).isNull();
+        }
+    }
+
+    private void clearSystemProperties() {
+        System.clearProperty(SdkSystemSetting.AWS_ACCESS_KEY_ID.property());
+        System.clearProperty(SdkSystemSetting.AWS_SECRET_ACCESS_KEY.property());
+        System.clearProperty(SdkSystemSetting.AWS_CREDENTIAL_SCOPE.property());
+    }
+
+    private ProfileFile credentialFile(String name, String accessKeyId, String secretAccessKey, String credentialScope) {
+        String contents = String.format("[%s]\naws_access_key_id = %s\naws_secret_access_key = %s\naws_credential_scope = %s",
+                                        name, accessKeyId, secretAccessKey, credentialScope);
```

Comment:
Is a blank `aws_credential_scope` a case we need to test? or is that treated the same as null for priority purposes?